pkg/master/master.go

```diff
@@ -301,6 +308,9 @@ func (m *Master) InstallAPIs(c *Config) {
 		if !enabled {
 			continue
 		}
+		if postHookProvider, ok := restStorageBuilder.(RESTStoragePostStartHookProvider); ok {
+			m.GenericAPIServer.PostStartHooks = append(m.GenericAPIServer.PostStartHooks, postHookProvider.PostStartHook(c.APIResourceConfigSource, restOptionsGetter))
```

Comment:
Add a method to generic apiserver for this. That way you can 
* lock in case people add them from multiple threads
* return an error if post start hooks have already been called